aten/src/ATen/native/sparse/SparseBinaryOpIntersectionCommon.h

```diff
@@ -146,10 +148,23 @@ void _sparse_binary_op_intersection_kernel_impl(
       " to output ", res.scalar_type());
 
   using KernelLauncher = KernelLauncher<kernel_t>;
+  using OptTensor = c10::optional<Tensor>;
 
   // If the op and sum are not distributive, coalesce is required.
-  const Tensor x = distributive_with_sum ? x_ : x_.coalesce();
-  const Tensor y = distributive_with_sum ? y_ : y_.coalesce();
+  const auto coalesce_if_not_distributive = [distributive_with_sum](const Tensor& t, const OptTensor& t_hash_opt) -> auto {
+    // No need to coalesce in such a case.
+    if (distributive_with_sum) {
+      return std::make_tuple(t, t_hash_opt);
+    } else {
+      // Otherwise coalesce and force hash recompute.
+      return std::make_tuple(t.coalesce(), static_cast<OptTensor>(c10::nullopt));
+    }
```

Comment:
TODO: no need to recompute, just sufficient to wrap into a sparse tensor and coalesce.